src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -517,10 +515,28 @@ UCS_CLASS_INIT_FUNC(uct_rc_mlx5_iface_common_t,
         goto cleanup_tm;
     }
 
-    self->super.config.fence       = uct_ib_device_has_pci_atomics(dev);
+    self->super.config.fence       = rc_config->fence != UCT_RC_FENCE_NONE;
     self->super.rx.srq.quota       = self->rx.srq.mask + 1;
     self->super.config.exp_backoff = mlx5_config->exp_backoff;
 
+    switch (self->super.config.fence) {
```

Comment:
how come we are doing switch/case here if according to line 518 config.fence can be only 1 or 0 ??